### PR TITLE
Feat: Implements #297 request

### PR DESCRIPTION
**Pull request by Shubhcoder**
### Reasons for making this change

 I have added the code (with tests) to implement the feature requested in #297.  Also, I have refactored the duplicate (non-React component) based function down into `@rjsf/utils` under the file name `getDateElementProps.ts`.

### Checklist

- [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [x] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [x] **I'm adding a new feature**
  - [x] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/utils/src/getDateElementProps.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+export type DateElementFormat = 'DMY' | 'MDY' | 'YMD';
+
+type DateElementProp = {
+  type: string;
+  range: [number, number];
+  value: number | undefined;
+};
+
+export default function getDateElementProps(
```
Comment (heath-freenome) [inline packages/utils/src/getDateElementProps.ts:11]: Can you please add documentation to this function as well as replicating that documentation into the official RJSF online documentation found here `packages/docs/docs/api-reference/utility-functions.md`

**Review by heath-freenome — CHANGES_REQUESTED** [review]
@Shubhcoder Thanks for doing this! Can you update the `CHANGELOG.md` file as well with a description of this feature? I would expect to see sections for `@rjsf/antd`, `@rjsf/chakra-ui`, `@rjsf/core` and a `Dev / Docs / Playground` section.

File: packages/utils/src/getDateElementProps.ts (hunk 1 above)
Comment (Shubhcoder) [reply packages/utils/src/getDateElementProps.ts:22]: Hi, @heath-freenome 

I have updated my PR with the required changes.

File: packages/docs/docs/api-reference/utility-functions.md [hunk 2]
```diff
@@ -281,6 +281,21 @@ Returns `undefined` when a valid discriminator is not present.
 
 - string | undefined: The `discriminator.propertyName` if it exists in the schema, otherwise `undefined`
 
+### getDateElementProps( date: DateObject, time: boolean, yearRange: [number, number] = [1900, new Date().getFullYear() + 2], format: DateElementFormat = 'YMD')
```
Comment (heath-freenome) [inline packages/docs/docs/api-reference/utility-functions.md:284]: We don't need to mention the parameters here as they are explicitly mentioned below
```suggestion
### getDateElementProps()
```

**Review by heath-freenome — APPROVED** [review]

**Comment by heath-freenome** [comment]
@Shubhcoder I'm not sure why the mainline merge is failing your builds now.  Can you please investigate?

File: packages/core/src/components/widgets/AltDateWidget.tsx [hunk 3]
```diff
@@ -5,11 +5,13 @@ import {
   toDateString,
   pad,
   DateObject,
+  type DateElementFormat,
```
Comment (heath-freenome) [inline packages/core/src/components/widgets/AltDateWidget.tsx:8]: I'm not sure the `type` is needed?
```suggestion
  DateElementFormat,
```

**Pull request merged**